### PR TITLE
Change Request 1

### PR DESCRIPTION
**Pull request by LeticiaSan**

### CODE REVIEW

File: src/WelcomeServices.java [hunk 1]
```diff
@@ -7,15 +7,24 @@
...
 		public Map<String, User> mapEmail = new HashMap<String, User>();
-				
+		public boolean enableSearchByEmail;
+
+		public void inputEnableEmail(String s){
+			this.enableSearchByEmail = parseBoolean(s);
+		}
+
+		public static boolean parseBoolean(String s){
+			return s.equals("true") || s.equals("1");
```
Comment (oswaldobapvicjr) [inline src/WelcomeServices.java:40]: Probable NullPointerException raiser here. Either handle or document it.

File: src/Welcome.java [hunk 2]
```diff
@@ -4,20 +4,22 @@
 		
 	public static void main(String[] args) {
 		WelcomeServices welcomeServices = new WelcomeServices();
-		
+		String enableSearchByEmail = "";
```
Comment (oswaldobapvicjr) [inline src/Welcome.java]: This should be loaded from the welcome.properties file, according to the CR requirement.

File: src/WelcomeServices.java [hunk 3]
```diff
@@ -51,7 +60,7 @@ public String searchUserByCpf(String inputCpf){
 		
 		public String searchUserByEmailAdress(String inputEmail){
 			User user = this.mapEmail.get(inputEmail);
-			if(user != null) {
+			if(user != null && enableSearchByEmail) {
```
Comment (oswaldobapvicjr) [inline src/WelcomeServices.java:86]: Please re-evaluate this metho. You are searching the user even if the search by email is disabled. Can you explain why this is not a good idea?

**Review by oswaldobapvicjr — CHANGES_REQUESTED** [review]

File: src/WelcomeServices.java (hunk 3 above)
Comment (LeticiaSan) [reply src/WelcomeServices.java:86]: it's an unnecessary search, slowing down the program. I already fixed it,

**Pull request closed without merging**